spa/src/pages/ConfigSteps/index.tsx

```diff
@@ -55,31 +56,82 @@ const InlineDialog = styled(TooltipPrimitive)`
 	position: absolute;
 	top: -22px;
 `;
-
+const LoggedInContent = styled.div`
+	display: flex;
+	justify-content: center;
+	align-items: center;
+`;
 const ConfigSteps = () => {
-	const [completedStep1, setCompletedStep1] = useState(false);
+	const { username, email } = OAuthManagerInstance.getUserDetails();
+	const isAuthenticated = !!(username && email);
+
+	const originalUrl = window.location.origin;
+	const [selectedOption, setSelectedOption] = useState(0);
+
+	const [completedStep1, setCompletedStep1] = useState(isAuthenticated);
 	const [completedStep2] = useState(false);
+
 	const [showStep2, setShowStep2] = useState(true);
-	const [canViewContentForStep2, setCanViewContentForStep2] = useState(false);
-	const [selectedOption, setSelectedOption] = useState(0);
+	const [canViewContentForStep2, setCanViewContentForStep2] = useState(isAuthenticated);
+
+	const [expandStep1, setExpandStep1] = useState(!isAuthenticated);
+	const [expandStep2, setExpandStep2] = useState(isAuthenticated);
 
-	const authorize = () => {
+	const [isLoggedIn, setIsLoggedIn] = useState(isAuthenticated);
+	const [loggedInUser, setLoggedInUser] = useState<string>(username);
+	const [loaderForLogin, setLoaderForLogin] = useState(false);
+
+	useEffect(() => {
+		window.addEventListener("message", (event) => {
```

Comment:
Do you think we should clear the eventListener as well?

Reply:
Removed the eventListener once the message has been received.